haystack/telemetry.py

```diff
@@ -146,14 +146,18 @@ def send_pipeline_event(  # type: ignore
                 telemetry.send_event(event_name="Public Demo", event_properties=event_properties)
                 return
 
-            # Send this event only if the pipeline config has changed
-            if pipeline.last_config_hash == pipeline.config_hash:
+            # If pipeline config has not changed, send an event every 100 runs
+            if pipeline.last_config_hash == pipeline.config_hash and pipeline.runs % 100 == 0:
```

Comment:
I would put the 100 in a constant, so it's safe to change later if we need to adjust the events rate